theano/misc/pycuda_example.py

```diff
@@ -285,6 +285,11 @@ def __init__(self, scalar_op, inplace_pattern=None, name=None):
         self.scalar_op = scalar_op
         self.inplace_pattern = inplace_pattern
 
+    # As we have a dict in props, we need to implement __hash__
+    def __hash__(self):
+        return (hash(type(self)) ^ hash(self.scalar_op) ^
+                hash_from_dict(self.inplace_pattern))
```

Comment:
I'm not sure I understand. I was suggesting to update the parent `__hash__` method so it calls `hash_from_dict` on the elements of `__props__` that are dicts. How do the hash collisions for ints relate to that? Also, collisions in hashes are not necessarily a problem.
I don't like the idea of just applying xor on the different hash elements in general, because xor it is commutative.